### PR TITLE
Added New component: LinearProgress

### PR DESCRIPTION
**Pull request by arpitBhalla**
**What kind of change does this PR introduce?**

New Component ```LinearProgress``` 
According to material desgin guidelines https://material.io/components/progress-indicators#linear-progress-indicators

**Did you add tests for your changes?**
Yupp

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**
Nope

**Other information**

### CODE REVIEW

**Comment by arpitBhalla** [comment]
@flyingcircle Please Review, I havn't exported this from index.ts, tests are pending, **Please Share your views**

**Comment by arpitBhalla** [comment]
https://user-images.githubusercontent.com/55053424/111347508-4a58ee00-86a5-11eb-9c69-57c829c65ddb.mp4

File: src/linearProgress/LinearProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { View, StyleProp, ViewStyle, Animated, Platform } from 'react-native';
+import { withTheme, FullTheme } from '../config';
+import Color from 'color';
+
+export type LinearProgressProps = {
+  value?: number; // 0 to 1
+  variant?: 'determinate' | 'indeterminate';
+  color?: 'primary' | 'secondary';
```
Comment (flyingcircle) [inline src/linearProgress/LinearProgress.tsx:9]: Color being limited to only the primary or secondary color seems overly restrictive. People will want to be able to style this however they want.

**Review by flyingcircle — CHANGES_REQUESTED** [review]
This looks awesome! I think the component should probably also extend `ViewProps`. You'll definitely want people to be able to extend certain capabilities such as `accessibility` to the component.

File: src/linearProgress/LinearProgress.tsx (hunk 1 above)
Comment (flyingcircle) [reply src/linearProgress/LinearProgress.tsx:9]: Perhaps just default to the secondary color otherwise allow people to override the color.

Comment (arpitBhalla) [reply src/linearProgress/LinearProgress.tsx:9]: Sure Sir! Will do that

**Comment by arpitBhalla** [comment]
@flyingcircle Kindly review

**Comment by flyingcircle** [comment]
Code looks good just looking through it. I'll try to test it out myself here later on in a bit.

**Comment by arpitBhalla** [comment]
@flyingcircle sure sir🙂

**Comment by flyingcircle** [comment]
So I tested it out in iOS using your code. I made this adjustment:
```js
        <Text>Linear Progress Indeterminate </Text>
        <LinearProgress style={{ marginVertical: 10 }} color="red" />
        <Text>Linear Progress Secondary color</Text>
        <LinearProgress style={{ marginVertical: 10 }} color="secondary" />
        <Text>Linear Progress with value</Text>
        <LinearProgress
          style={{ marginVertical: 10 }}
          value={progress}
          variant="determinate"
          color="red"
        />
        <Text>Linear Progress with value secondary color</Text>

        <LinearProgress
          style={{ marginVertical: 10 }}
          value={progress}
          variant="determinate"
          color="secondary"
        />
        <Button
          disabled={progress > 0}
          onPress={() => {
            setProgress(0.00001);
          }}
          title={'Start Progress'}
          containerStyle={{ margin: 10 }}
        />
        <Button
          disabled={progress === 0}
          onPress={() => {
            setProgress(0);
          }}
          title={'Restart'}
          containerStyle={{ margin: 10 }}
        />
```

I wasn't able to see the progress in the `red` bars when testing with the demo app on iOS. You currently have it where the Color is like a "dark pink" on "lighter pink". I think it may also be reasonable to have a `trackColor` prop so that they aren't forced into using the same color for both.

**Comment by arpitBhalla** [comment]
@flyingcircle I made Changes, Please Review

**Comment by flyingcircle** [comment]
Awesome! This looks great! I only see only minor change that should be made to the demo, but I'll post it there. Great work!

**Comment by arpitBhalla** [comment]
@flyingcircle Thanku Sir

**Pull request merged**